src/components/WebGLUpload.tsx

```diff
@@ -0,0 +1,361 @@
+        throw new Error("Failed to upload file after 4 retries");
+      }
+    }),
+  );
+
+  await axios.put(
+    `/api/games/${gameId}`,
+    JSON.stringify({ webGLBuild: true }),
+    {
+      headers: {
+        "Content-Type": "text",
+      },
+    },
+  );
+}
+
+interface Props {
+  cancel: () => void;
+  setLoaderFile: React.Dispatch<React.SetStateAction<null | File>>;
+  setDataFile: React.Dispatch<React.SetStateAction<null | File>>;
+  setCodeFile: React.Dispatch<React.SetStateAction<null | File>>;
+  setFrameworkFile: React.Dispatch<React.SetStateAction<null | File>>;
+  setUploadedWebGL: React.Dispatch<React.SetStateAction<boolean>>;
+  setUploadedFilenames: React.Dispatch<React.SetStateAction<string[]>>;
+  setShowUploadGameBuild: React.Dispatch<React.SetStateAction<boolean>>;
+  setShowUploadedBuild: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+export default function WebGLUpload({
+  cancel,
+  setLoaderFile,
+  setDataFile,
+  setCodeFile,
+  setFrameworkFile,
+  setUploadedWebGL,
+  setUploadedFilenames,
+  setShowUploadGameBuild,
+  setShowUploadedBuild,
+}: Props) {
+  const [loaderName, setLoaderName] = useState<null | string>(null);
+  const [dataName, setDataName] = useState<null | string>(null);
+  const [codeName, setCodeName] = useState<null | string>(null);
+  const [frameworkName, setFrameworkName] = useState<null | string>(null);
+  // const [gameId, setGameId] = useState("");
+
+  const handleFileChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (event.target.files === null || event.target.files.length === 0) return;
+    const file = event.target.files[0];
+    const name = event.target.name;
+
+    switch (name) {
+      case "loader":
+        setLoaderFile(file);
+        setLoaderName(file.name);
+        break;
+      case "data":
+        setDataFile(file);
+        setDataName(file.name);
+        break;
+      case "code":
+        setCodeFile(file);
+        setCodeName(file.name);
+
+        break;
+      case "framework":
+        setFrameworkFile(file);
+        setFrameworkName(file.name);
+
+        break;
+      default:
+        break;
+    }
+  };
+
+  const handleSubmit = async () => {
+    // if (
+    //   loaderFile === null ||
+    //   dataFile === null ||
+    //   codeFile === null ||
+    //   frameworkFile === null
+    // ) {
+    //   alert("Please input all files");
+    //   return;
+    // }
+    // const files = new Map([
+    //   ["loader", loaderFile],
+    //   ["data", dataFile],
+    //   ["code", codeFile],
+    //   ["framework", frameworkFile],
+    // ]);
+    // try {
+    //   await uploadBuildFiles(gameId, files);
+    //   alert("Files uploaded successfully");
+    // } catch (e) {
+    //   console.error(e);
+    //   alert("Failed to upload files");
+    // }
+
+    if (
+      loaderName === null ||
+      dataName === null ||
+      codeName === null ||
+      frameworkName === null
+    ) {
+      alert("Please input all files");
```

Comment:
Instead of using alert, please notify the user following the [Figma design](https://www.figma.com/file/9qcNl3ZnyTQqnIcXMxiFOw/Jennifer-Annes-S24?type=design&node-id=12372-16574&mode=design&t=jCKVlKRmNjTWPYuv-4) -- same for everywhere else you use this alert